spec/schema/schema_dumper_spec.rb

```diff
@@ -0,0 +1,69 @@
+describe "SchemaDumper" do
+  SCHEMA_RB_FILE  = File.expand_path("../dummy/db/schema.rb", __dir__)
+  TABLE_LIST_FILE = File.expand_path("../support/table_list.txt", __dir__)
+
+  let(:combined_metric_list) { metric_rollup_list + metric_list }
+  let(:metric_list)          { (0..23).to_a.map { |num| "metrics_#{'%<num>02d' % { :num => num }}" } }
+  let(:metric_rollup_list)   { (1..12).to_a.map { |num| "metric_rollups_#{'%<num>02d' % { :num => num }}" } }
+  let(:table_list)           { File.read(TABLE_LIST_FILE).lines.map(&:strip) }
+  let(:schema)               { File.read(SCHEMA_RB_FILE) }
+
+  it "includes all of the manageiq tables" do
+    create_table_lines = schema.lines.grep(/^\s*create_table/)
+
+    expect(create_table_lines.count).to eq(table_list.count)
+
+    table_list.each do |table_name|
+      has_table = create_table_lines.detect { |l| l.include?(table_name) }
+      expect(has_table).to be_truthy
```

Comment:
Super minor, but in my experience, expects in a loop become hard to debug because when it blows up all this will say is "expected false to be true" and you have no idea _which_ table_name was the one responsible.  There are 2 general ways to fix this

1) Add an extra message like `expect(has_table).to be_truthy, "table not found for #{table_name}"
2) Put the loop *outside* of the `it`, and have a separate `it` per table_name, with the table name as part of the `it` description.

Reply:
Will probably go with option 1, since it is less "meta-program-y", and at least the error output will be more obvious.

* * *

Sidenote:  The reason I am doing the `.detect` prior is that when trying to do a lookup prior to this against the entire schema string, the `diff` from `rspec` is basically useless, so this is a hopefully slightly more legible solution.